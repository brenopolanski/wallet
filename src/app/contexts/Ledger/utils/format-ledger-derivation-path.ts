import { LedgerDerivationScheme } from "app/contexts/Ledger/contracts";

export const formatLedgerDerivationPath = (scheme: LedgerDerivationScheme) =>
	`m/${scheme.purpose || 44}'/${scheme.coinType}'/${scheme.account || 0}'/${scheme.change || 0}/${
		scheme.address || 0
	}`;
