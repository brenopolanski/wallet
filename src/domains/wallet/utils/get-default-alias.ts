import { Networks } from "@payvo/sdk";
import { Contracts } from "@payvo/sdk-profiles";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${network.displayName()} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let sameCoinWalletsCount = sameCoinWallets.length || 1;

	while (profile.wallets().findByAlias(makeAlias(sameCoinWalletsCount))) {
		sameCoinWalletsCount++;
	}

	return makeAlias(sameCoinWalletsCount);
};
