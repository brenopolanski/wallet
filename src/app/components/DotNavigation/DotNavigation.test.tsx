import userEvent from "@testing-library/user-event";
import React from "react";
import { render } from "utils/testing-library";

import { DotNavigation } from "./DotNavigation";

describe("DotNavigation", () => {
	it("should render", () => {
		const { container, asFragment } = render(<DotNavigation />);

		expect(container).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	it("handles click on a dot", () => {
		const clickMock = jest.fn();
		const { getByTestId } = render(<DotNavigation onClick={clickMock} />);

		userEvent.click(getByTestId("DotNavigation-Step-1"));

		expect(clickMock).toHaveBeenCalledWith(1);
	});
});
