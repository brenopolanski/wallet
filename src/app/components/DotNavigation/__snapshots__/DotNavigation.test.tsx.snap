// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DotNavigation should render 1`] = `
<DocumentFragment>
  <ul
    class="DotNavigation__DotNavigationWrapper-sc-q7l2tr-1 eJDorE"
    data-testid="DotNavigation"
  >
    <li
      class="DotNavigation__StepStyled-sc-q7l2tr-0 fldKra"
      data-testid="DotNavigation-Step-0"
    />
    <li
      class="DotNavigation__StepStyled-sc-q7l2tr-0 cFVEMI"
      data-testid="DotNavigation-Step-1"
    />
  </ul>
</DocumentFragment>
`;
